packages/accordions/src/elements/accordion/Accordion.tsx

```diff
@@ -0,0 +1,118 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+import React, {
+  useRef,
+  useEffect,
+  forwardRef,
+  RefAttributes,
+  HTMLAttributes,
+  PropsWithoutRef,
+  ForwardRefExoticComponent
+} from 'react';
+import { useAccordion } from '@zendeskgarden/container-accordion';
+import { StyledAccordion } from '../../styled';
+import { AccordionContext } from '../../utils';
+import { Section } from '../accordion/components/Section';
+import { Header } from '../accordion/components/Header';
+import { Label } from '../accordion/components/Label';
+import { Panel } from '../accordion/components/Panel';
+
+interface IStaticAccordionExport<T, P>
+  extends ForwardRefExoticComponent<PropsWithoutRef<P> & RefAttributes<T>> {
+  Section: typeof Section;
+  Header: typeof Header;
+  Label: typeof Label;
+  Panel: typeof Panel;
+}
+
+interface IAccordionProps extends Omit<HTMLAttributes<HTMLDivElement>, 'onChange'> {
+  /** The `aria-level` used to apply heading rank on accordion headers */
+  level: number;
```

Comment:
Is this a practice we typically do? To take an aria attribute available to the element and make it part of the API? Could we advise this attribute to be added to to the component using the actual aria attribute name?

Additionally, I'm curious, is there any way possible we could take care of this programmatically somehow based on children qty?

Reply:
On the other hand, I'm seeing now it's required so it's sort of nice to control the accessibility implementation in that way, my concern is it might set a precedent we could apply to most components where we start making aria attributes mandatory in the API.

Reply:
> Controlled usage allows consumers to have full control of the accordion state and drive custom behavior (such as expand all, or expanding a single accordion panel).

This was my initial approach, however @austin94 pointed out some problems* with adding heading rank to the accordion headers programmatically. Austin proposed the current implementation of the `level` prop as most suitable, and I agree that it was much better than the original (programmatic solution) that I had.

> my concern is it might set a precedent we could apply to most components where we start making aria attributes mandatory in the API.

I agree! I think most accessibility should be abstracted away from the consumer. [These](https://gist.github.com/ryanflorence/e5c794e6093d16a69fa88d2112a292f7#accessible) are some of the person guiding principles I follow. I think this particular prop is a special case, and should not set a precedent of not abstracting away accessibility when possible.

_* The problem with programmatically determining header rank is that rank can sometimes depend on the context of the content. It's not a sure thing that an `h2` rank belongs under an `h1` rank. This was difficult for me to grok, and Austin had to go to school on me to get this across my head haha._

Reply:
> is there any way possible we could take care of this programmatically somehow based on children qty

This prop isn't telling us how many children are within the component. It's telling us where the component falls within the ancestoral DOM hierarchy. This is information we can't know, so we need the user to tell us :wink:

Reply:
Yes yes yes, I see now. Very awesome.